e2e-tests/cypress/e2e/journeys/insurance/your-business/company-details/validation/company-details-invalid-phone-number-validation.spec.js

```diff
@@ -0,0 +1,204 @@
+          .then((text) => {
+            expect(text.trim()).equal(errorMessage);
+          });
+        companyDetails.phoneNumberError().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(`Error: ${errorMessage}`);
+          });
+      });
+    });
+
+    describe('landline number with too few numbers', () => {
+      it('should display validation errors', () => {
+        companyDetails.companiesHouseSearch().clear().type(COMPANIES_HOUSE_NUMBER);
+        yesRadioInput().eq(0).click();
+        yesRadioInput().eq(1).click();
+        companyDetails.companyWebsite().clear().type(WEBSITE_EXAMPLES.VALID);
+        companyDetails.phoneNumber().clear().type(INVALID_PHONE_NUMBERS.LANDLINE_SHORT);
+        submitButton().click();
+        partials.errorSummaryListItems().should('have.length', 1);
+        partials.errorSummaryListItems().first().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(errorMessage);
+          });
+        companyDetails.phoneNumberError().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(`Error: ${errorMessage}`);
+          });
+      });
+    });
+
+    describe('special characters in phone number', () => {
+      it('should display validation errors', () => {
+        companyDetails.companiesHouseSearch().clear().type(COMPANIES_HOUSE_NUMBER);
+        yesRadioInput().eq(0).click();
+        yesRadioInput().eq(1).click();
+        companyDetails.companyWebsite().clear().type(WEBSITE_EXAMPLES.VALID);
+        companyDetails.phoneNumber().clear().type(INVALID_PHONE_NUMBERS.LANDLINE_SPECIAL_CHAR);
+        submitButton().click();
+        partials.errorSummaryListItems().should('have.length', 1);
+        partials.errorSummaryListItems().first().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(errorMessage);
+          });
+        companyDetails.phoneNumberError().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(`Error: ${errorMessage}`);
+          });
+      });
+    });
+
+    describe('letters in phone number', () => {
+      it('should display validation errors', () => {
+        companyDetails.companiesHouseSearch().clear().type(COMPANIES_HOUSE_NUMBER);
+        yesRadioInput().eq(0).click();
+        yesRadioInput().eq(1).click();
+        companyDetails.companyWebsite().clear().type(WEBSITE_EXAMPLES.VALID);
+        companyDetails.phoneNumber().clear().type(INVALID_PHONE_NUMBERS.LANDLINE_LETTER);
+        submitButton().click();
+        partials.errorSummaryListItems().should('have.length', 1);
+        partials.errorSummaryListItems().first().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(errorMessage);
+          });
+        companyDetails.phoneNumberError().invoke('text')
+          .then((text) => {
+            expect(text.trim()).equal(`Error: ${errorMessage}`);
+          });
```

Comment:
Finding these tests a bit hard to read, could we add some spacing? Also could we extract the first few lines of each test? Seems to be repeated a lot, something like this?

```js
const completeAllOtherFields = () => {
  companyDetails.companiesHouseSearch().clear().type(COMPANIES_HOUSE_NUMBER);
  yesRadioInput().eq(0).click();
  yesRadioInput().eq(1).click();

  companyDetails.companyWebsite().clear().type(WEBSITE_EXAMPLES.VALID);  
};


it('should...', () => {
  completeAllOtherFields();

  companyDetails.phoneNumber().clear().type(INVALID_PHONE_NUMBERS.LANDLINE_LETTER);
  submitButton().click();

  partials.errorSummaryListItems().should('have.length', 1);
  partials.errorSummaryListItems().first().invoke('text')
    .then((text) => {
      expect(text.trim()).equal(errorMessage);
    });

  companyDetails.phoneNumberError().invoke('text')
    .then((text) => {
      expect(text.trim()).equal(`Error: ${errorMessage}`);
    });
});
```

Reply:
done